lib/process_settings/abstract_monitor.rb

```diff
@@ -94,10 +96,24 @@ def static_context=(context)
     def targeted_value(*path, dynamic_context:, required: true)
       # Merging the static context in is necessary to make sure that the static context isn't shifting
       # this can be rather costly to do every time if the dynamic context is not changing
-      # TODO: Warn in the case where dynamic context was attempting to change a static value
-      # TODO: Cache the last used dynamic context as a potential optimization to avoid unnecessary deep merges
-      # TECH-4402 was created to address these todos
-      full_context = dynamic_context.deep_merge(static_context)
+
+      # Warn in the case where dynamic context was attempting to change a static value
+      dynamic_context.each do |key, value|
```

Comment:
This look assumes a few things:
1. that the dynamic and static context hashes are flat structures
2. that we want to alert to only the first encountered conflict
3. that we want to alert even if the dynamic context contains the same value

Lets work with @ColinDKelley to see if we can't come up with an efficient algorithm that will:
1. scan the full structure of the static context
2. only alert to keys that are causing a change
3. alert to all the full paths that were attempted to be overridden by the dynamic context

Reply:
Good catch, James.

The general approach for these algorithms is:
- Accumulate everything that has changed into a "change set".
- If the change set isn't empty, alert, including the change set.

Reply:
BTW I don't think we'll need a recursive algorithm, since hashes can be compared for equality.

So it'll be something like:
```
changes = dynamic_context.each_with_object({}) do |(key, dynamic_value), result|
  if static_context.has_key?(key)
    static_value = static_context[key]
    if static_value != dynamic_value
      result[key] = [static_value, dynamic_value]
    end
  end
end

changes.empty? or warn("WARNING: static context overwritten by dynamic!\n#{changes.inspect}")
```

Reply:
This is a clean approach! I'll implement these changes 👍